enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/api/Utils.java

```diff
@@ -90,6 +91,11 @@ static void populateRemoveAndProtectedHeaders(RequestContext requestContext) {
         // generic OPTIONS method call happens, matchedResourcePath becomes null.
         if (requestContext.getMatchedResourcePath() != null &&
                 requestContext.getMatchedResourcePath().isDisableSecurity()) {
+            // If the resource has disabled security, and  if the connection request
+            // is from the Choreo-console Test console UI,  it is required to add
+            // choreo-internal-API-Key header to the response header x-websocket-protocol
+            InternalAPIKeyUtils.addWSProtocolResponseHeaderIfRequired(requestContext);
```

Comment:
This function `populateRemoveAndProtectedHeaders` prevents protected headers from being passed to the backend. Shall we move this function call to the method `authenticate` in the class `UnsecuredAPIAuthenticator`?

Reply:
And shall we check for the API type? WS or Rest API